tests/test-http-api/test-version.js

```diff
@@ -0,0 +1,52 @@
+/* globals describe, it */
+
+'use strict'
+
+const expect = require('chai').expect
+const APIctl = require('ipfs-api')
+
+describe('version', () => {
+  describe('api', () => {
+    var api
+
+    it('api', (done) => {
+      api = require('../../src/http-api').server.select('API')
+      done()
+    })
+
+    it('get the version', (done) => {
+      api.inject({
+        method: 'GET',
+        url: '/api/v0/version'
+      }, res => {
+        expect(res.result.Version).to.equal('0.4.0-dev')
+        expect(res.result).to.have.a.property('Commit')
+        expect(res.result).to.have.a.property('Repo')
+        done()
+      })
+    })
+  })
+
+  describe('gateway', () => {
+    // TODO
+  })
+
+  describe('using js-ipfs-api', () => {
+    var ctl
+
+    it('start IPFS API ctl', (done) => {
+      ctl = APIctl('/ip4/127.0.0.1/tcp/6001')
+      done()
+    })
+
+    it('get the version', (done) => {
+      ctl.version((err, result) => {
```

Comment:
@dignifiedquire I'm hitting an issue here that I would love to have your review. I implemented the http-api endpoint following the spec and you can see the test above (line 17) that it is indeed callable and returns the data as expected, however, the problem is that this js-ipfs-api call is giving me the result as a 'buffer', instead of the object correctly parsed, my first intuition was it was lacking headers, but I confirmed that I was adding the same headers as go-ipfs and still I got nothing. Any ideas? Thank you.